packages/playwright-test/src/matchers/toMatchSnapshot.ts

```diff
@@ -145,6 +145,7 @@ class SnapshotHelper<T extends ImageComparatorOptions> {
       maxDiffPixels: options.maxDiffPixels,
       maxDiffPixelRatio: options.maxDiffPixelRatio,
       threshold: options.threshold,
+      comparator: options.comparator,
```

Comment:
I think you should update the `new SnapshotHelper()` call in `toHaveScreenshot` to include `comparator` option.